.github/workflows/wave.yml

```diff
@@ -0,0 +1,87 @@
+name: Generate conda-lock
+# When environment.yml is changed
+on:
+  pull_request:
+    paths:
+      - "**/environment.yml"
+
+jobs:
+  gen-matrix:
+    name: generate-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+
+      - name: Calculate file differences
+        id: diff
+        uses: tj-actions/changed-files@v44
+        with:
+          json: true
+          quotepath: false
+          files: |
+            modules/**/environment.yml
+      - name: Debug
+        run: echo ${{ steps.diff.outputs.all_changed_files }}
+      - id: set-matrix
+        run: echo "matrix={\"files\":${{ steps.diff.outputs.all_changed_files }} }" >> "$GITHUB_OUTPUT"
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  lock:
+    if: "${{ fromJson(needs.gen-matrix.outputs.matrix) }}"
+    needs: gen-matrix
+    name: Generate conda-lock file
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: "${{ fromJson(needs.gen-matrix.outputs.matrix) }}"
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: conda-lock-env
+          create-args: >-
+            python=3.11
+            conda-lock
+            mamba
+
+      - name: Get the lock file path
+        uses: actions/github-script@v7
+        id: lockfile-name
+        with:
+          result-encoding: string
+          script: |
+            return '${{ matrix.files }}'.replace('/environment.yml', '/conda-lock.yml');
+
+      - name: Run conda-lock
+        run: |
+          rm --force "${{steps.lockfile-name.outputs.result}}"
+          conda-lock lock $MAMBA --file "${{ matrix.files }}" --kind lock --platform linux-64 --lockfile "${{steps.lockfile-name.outputs.result}}"
```

Comment:
So, I tested why the lock file didn't work, and I think it's because of the `--platform` here. To get a lock file compatible with Nextflow (which just does [like this](https://github.com/nextflow-io/nextflow/blob/5eeed5bf096c05d77c79863a6d9e48d09cdd5fcc/modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy#L280-L297)), you have to render it to a platform specific one:

```
conda-lock render -p osx-64  
Rendering lockfile(s) for osx-64...
 - Install lock using : conda create --name YOURENV --file conda-osx-64.lock
```

So for nextflow as-is, we will need platform-specific lockfiles.